api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineConfigurationManagerTest.java

```diff
@@ -16,182 +16,155 @@
-        dataStorage.setMountOptions("testMountOptions2");
-        dataStorage.setOwner(TEST_OWNER2);
-        dataStorageDao.createDataStorage(dataStorage);
-        dataStorages.add(dataStorage);
-
-        bucketDataStorage = new S3bucketDataStorage(dataStorageDao.createDataStorageId(),
-                                                                        "testBucket2", "test/path4");
-        bucketDataStorage.setOwner(TEST_OWNER2);
-        dataStorageDao.createDataStorage(bucketDataStorage);
-        dataStorages.add(bucketDataStorage);
-
-        dataStorages.forEach(ds -> aclTestDao.createAclForObject(ds));
-        aclTestDao.grantPermissions(dataStorage, TEST_OWNER1,
-                                    Collections.singletonList((AclPermission) AclPermission.READ));
+        MockitoAnnotations.initMocks(this);
+        dataStorages.add(getNfsDataStorage(NFS_ID_1, "test/path1", "options1", "/some/path", OWNER1));
+        dataStorages.add(getS3bucketDataStorage(S3_ID_1, "test/path2", OWNER1));
+        dataStorages.add(getNfsDataStorage(NFS_ID_2, "test/path3", "options2", "", OWNER2));
+        dataStorages.add(getS3bucketDataStorage(S3_ID_2, "test/path4", OWNER2));
     }
 
     @Test
-    @Transactional(propagation = Propagation.REQUIRES_NEW)
-    @WithMockUser(username = TEST_OWNER1)
-    public void testGetUnregisteredPipelineConfiguration() {
-        PipelineStart vo = getPipelineStartVO();
-        PipelineConfiguration config = pipelineConfigurationManager.getPipelineConfiguration(vo);
-        Assert.assertFalse(config.getBuckets().isEmpty());
-        Assert.assertFalse(config.getNfsMountOptions().isEmpty());
+    public void shouldGetUnregisteredPipelineConfiguration() {
+        doReturn(TEST_DOCKER_IMAGE).when(pipelineVersionManager).getValidDockerImage(eq(TEST_IMAGE));
+        doReturn(dataStorages).when(dataStorageApiService).getWritableStorages();
 
-        String[] buckets = config.getBuckets().split(";");
+        final PipelineStart vo = getPipelineStartVO();
+        final PipelineConfiguration config = pipelineConfigurationManager.getPipelineConfiguration(vo);
+        assertFalse(config.getBuckets().isEmpty());
+        assertFalse(config.getNfsMountOptions().isEmpty());
 
-        Assert.assertEquals(2, buckets.length);
+        String[] buckets = config.getBuckets().split(";");
+        assertEquals(4, buckets.length);
         for (String bucket : buckets) {
-            Assert.assertTrue(dataStorages.stream().anyMatch(ds -> bucket.equals(ds.getPathMask())));
+            assertTrue(dataStorages.stream().anyMatch(ds -> bucket.equals(ds.getPathMask())));
         }
 
-
-        String[] nfsOptions = config.getNfsMountOptions().split(";");
-
-        Assert.assertEquals(1, nfsOptions.length);
+        String[] nfsOptions = deleteEmptyElements(config.getNfsMountOptions().split(";"));
+        assertEquals(2, nfsOptions.length);
         for (String option : nfsOptions) {
```

Comment:
Could you please apply the approach from comment above?

Reply:
+